### PR TITLE
Custom Url configuration

### PR DESCRIPTION
**Pull request by jamesmckenzie**
#1102 

* adds configuration options for apiBase, uploadsBase, connectBase

allows consumers of the library to easily override the base urls for the stripe api to allow for easier testing (for example, with the stripe-mock service)

### CODE REVIEW

File: src/Stripe.net/Infrastructure/Public/StripeConfiguration.cs [hunk 1]
```diff
@@ -40,5 +43,47 @@ public static void SetApiKey(string newApiKey)
         {
             _apiKey = newApiKey;
         }
+
+        internal static string GetApiBase()
+        {
+            if (string.IsNullOrEmpty(_apiBase))
+            {
+                _apiBase = "https://api.stripe.com/v1";
```
Comment (ob-stripe) [inline src/Stripe.net/Infrastructure/Public/StripeConfiguration.cs]: Rather than hardcoding the default value here, could you instead use a constant? E.g. `Urls.DefaultBaseUrl`.

File: src/Stripe.net/Infrastructure/Public/StripeConfiguration.cs [hunk 2]
```diff
@@ -40,5 +43,47 @@ public static void SetApiKey(string newApiKey)
...
+        public static void SetApiBase(string baseUrl)
+        {
+            _apiBase = baseUrl;
+        }
+
+        internal static string GetUploadsBase()
+        {
+            if (string.IsNullOrEmpty(_uploadsBase))
+            {
+                _uploadsBase = "https://uploads.stripe.com/v1";
```
Comment (ob-stripe) [inline src/Stripe.net/Infrastructure/Public/StripeConfiguration.cs]: Same here.

File: src/Stripe.net/Infrastructure/Public/StripeConfiguration.cs [hunk 3]
```diff
@@ -40,5 +43,47 @@ public static void SetApiKey(string newApiKey)
...
+        public static void SetUploadsBase(string baseUrl)
+        {
+            _uploadsBase = baseUrl;
+        }
+
+        internal static string GetConnectBase()
+        {
+            if (string.IsNullOrEmpty(_connectBase))
+            {
+                _connectBase = "https://connect.stripe.com";
```
Comment (ob-stripe) [inline src/Stripe.net/Infrastructure/Public/StripeConfiguration.cs]: And here.

File: src/Stripe.net/Infrastructure/Urls.cs [hunk 4]
```diff
@@ -14,7 +14,7 @@ internal static class Urls
 
         public static string Coupons => BaseUrl + "/coupons";
 
-        public static string Plans => BaseUrl + "/plans";
+        public static string Plans => BaseUrl + "/plans";
```
Comment (ob-stripe) [inline src/Stripe.net/Infrastructure/Urls.cs]: You have some unnecessary whitespace here.

**Comment by ob-stripe** [comment]
Hi @jamesmckenzie, thanks a lot for the PR! Overall this looks fairly solid to me. I left a few comments, could you take a look?

**Comment by jamesmckenzie** [comment]
Thanks for taking a look. I've made the requested changes - let me know if there's anything else

**Comment by jamesmckenzie** [comment]
and mistake fixed!

**Comment by ob-stripe** [comment]
Thanks @jamesmckenzie! This looks good to me. Can I bother you to squash the 3 commits into a single one before I pull this in?

**Comment by jamesmckenzie** [comment]
not a problem, done!

**Comment by ob-stripe** [comment]
Perfect! Pulling this in.
